domain/src/main/java/org/oppia/domain/analytics/EventStorageHelper.kt

```diff
@@ -0,0 +1,97 @@
+package org.oppia.domain.analytics
+
+import androidx.lifecycle.LiveData
+import org.oppia.app.model.EventLog
+import org.oppia.app.model.EventLogs
+import org.oppia.data.persistence.PersistentCacheStore
+import org.oppia.util.data.AsyncResult
+import org.oppia.util.data.DataProviders
+import org.oppia.util.logging.Logger
+import javax.inject.Inject
+import javax.inject.Singleton
+
+@Singleton
+class EventStorageHelper @Inject constructor(
+  private val cacheStoreFactory: PersistentCacheStore.Factory,
+  private val dataProviders: DataProviders,
+  private val logger: Logger
+) {
+  private val eventLogStore =
+    cacheStoreFactory.create("event_logs", EventLogs.getDefaultInstance())
+
+  fun addEventLog(eventLog: EventLog) {
+    eventLogStore.storeDataAsync(updateInMemoryCache = true) {
+      it.toBuilder().addEventLog(eventLog).build()
+    }.invokeOnCompletion {
+      it?.let {
+        logger.e(
+          "DOMAIN",
+          "Failed to store event log",
+          it
+        )
+      }
+    }
+  }
+
+  suspend fun getOptionalEvents(): MutableList<EventLog>{
+    var eventLogList: MutableList<EventLog> = mutableListOf()
+    val eventList = eventLogStore.readDataAsync().await().eventLogList
+    for (eventLog in eventList){
+      if(eventLog.priority == EventLog.Priority.OPTIONAL){
+        eventLogList.add(eventLog)
+      }
+    }
```

Comment:
You can do this in a more optimized way by using collection filters. See https://kotlinlang.org/docs/reference/collection-filtering.html

Reply:
Done, PTAL